### PR TITLE
Bump cython, numpy

### PR DESCRIPTION
**Pull request by nauaneed**

### CODE REVIEW

**Review by prabhuramachandran — CHANGES_REQUESTED** [review]
Looks good, thanks! Can you please bump the Python version to at least 3.11 or 3.12 so the tests run on latest Python.

File: compyle/types.py [hunk 1]
```diff
@@ -188,6 +188,7 @@ def _inject_types_in_module():
     NP_C_TYPE_MAP[np.dtype(np.uint64)] = 'unsigned long long'
     C_NP_TYPE_MAP['long long'] = np.int64
     C_NP_TYPE_MAP['unsigned long long'] = np.uint64
+    TYPES['long long'] = KnownType('long long')
```
Comment (nauaneed) [inline compyle/types.py:191]: This fixes tests failing on windows but not sure if this is a good fix. All keys in the `TYPES` dict do not contain spaces as they are meant to be used with `@annotate` decorator?

**Comment by nauaneed** [comment]
One failing test on windows remaining: `test_ext_module.py::TestExtModule::test_that_multiple_writes_do_not_occur_for_same_source`

**Review by prabhuramachandran — APPROVED** [review]
LGTM, thanks! We should fix the multiple writes error by using a proper cross platform lock like fasteners. Will try to do that later.

**Pull request merged**